advanced/Scripts/list.sh

```diff
@@ -131,28 +128,22 @@ AddDomain() {
         echo -e "  ${INFO} ${1} already exists in ${listname}, no need to add!"
       fi
     fi
-  elif [[ "${list}" == "${wildcardlist}" ]]; then
-    source "${piholeDir}/setupVars.conf"
-    # Remove the /* from the end of the IP addresses
-    IPV4_ADDRESS=${IPV4_ADDRESS%/*}
-    IPV6_ADDRESS=${IPV6_ADDRESS%/*}
+  elif [[ "${list}" == "${regexlist}" ]]; then
     [[ -z "${type}" ]] && type="--wildcard-only"
     bool=true
     # Is the domain in the list?
-    grep -e "address=\/${domain}\/" "${wildcardlist}" > /dev/null 2>&1 || bool=false
+    # Search only for exactly matching lines
+    grep -E "^${domain}$" "${regexlist}" > /dev/null 2>&1 || bool=false
```

Comment:
Will having a regex in regex mess things up? You could use `grep -F` to not interpret the regex input as regex.

Reply:
```
grep -F "${domain}" "${regexlist}"
```
wouldn't this also match if `$domain` is `abc` and `regex.list` contains `abcdefg`?

Reply:
Problem solved, I should read the `man` page...
```
-x
Select only those matches that exactly match the whole line.
```